src/Stratis.Bitcoin.Features.BlockStore/Repositories/BlockRepository.cs

```diff
@@ -454,25 +459,33 @@ public bool Exist(uint256 hash)
 
         protected virtual void OnDeleteTransactions(List<(Transaction, Block)> transactions)
```

Comment:
I think perhaps this and `OnDeleteBlocks` should no longer be virtual, especially because `OnDeleteBlocks` no longer calls this method. It gives the wrong impression about the intended usage/behaviour when subclassing.